sds_data_manager/lambda_code/SDSCode/opensearch_utils/snapshot.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import string
+from datetime import datetime
+
+import boto3
+import requests
+from requests_aws4auth import AWS4Auth
+
+
+def get_auth(region):
+    # Get AWS service and credentials for snapshot
+    service = "es"
+    credentials = boto3.Session().get_credentials()
+    awsauth = AWS4Auth(
+        credentials.access_key,
+        credentials.secret_key,
+        region,
+        service,
+        session_token=credentials.token,
+    )
+
+    return awsauth
+
+
+def register_repo(payload: dict, url: string, awsauth):
+    """Register the snapshot repo
+    Parameters
+    ----------
+    payload : dict
+             S3 bucket and AWS region to store the manual snapshots
+             The role ARN that has S3 permissions to store the new snapshot
+    url : str
+        OpenSearch domain URL endpoint including https:// and trailing /.
+    """
+
+    headers = {"Content-Type": "application/json"}
+
+    r = requests.put(url, auth=awsauth, json=payload, headers=headers)
+
+    return r
+
+
+def take_snapshot(url: string, awsauth):
+    """Initiate a new snapshot
+        Parameters
+    ----------
+    url : str
+        OpenSearch domain URL endpoint including https:// and trailing /.
+    """
+
+    r = requests.put(url, auth=awsauth)
+    return r
+
+
+def run_backup(host, region, snapshot_repo_name, snapshot_s3_bucket, snapshot_role_arn):
```

Comment:
Needs a docstring

Reply:
Yeah, looks like other functions needs update to their docstring.

Reply:
Agreed. I'll add the doctoring and update the others